src/components/Board.js

```diff
@@ -23,15 +36,15 @@ class Board extends Component {
       alert('Press Refresh')
     } else {
       if(this.state.board[index] === '') {
-        this.state.board[index] = this.state.currentTurn
+        let clonedBoard = this.state.board.slice(0)
+        clonedBoard[index] = this.state.currentTurn
         this.setState({
-          board: this.state.board,
+          board: clonedBoard,
           currentTurn: this.state.currentTurn === this.state.PLAYER_ONE_SYMBOL
           ? this.state.PLAYER_TWO_SYMBOL
           : this.state.PLAYER_ONE_SYMBOL,
-        })
+        }, this.checkForWinner)
```

Comment:
From React's documentation: 

setState() does not immediately mutate this.state but creates a pending state transition. Accessing this.state after calling this method can potentially return the existing value. There is no guarantee of synchronous operation of calls to setState and calls may be batched for performance gains.

Reply:
So I sent this.checkForWinner as a callback.